llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -4070,10 +4070,6 @@ InstructionCost AArch64TTIImpl::getShuffleCost(
         Cost += getShuffleCost(NumSources <= 1 ? TTI::SK_PermuteSingleSrc
                                                : TTI::SK_PermuteTwoSrc,
                                NTp, NMask, CostKind, 0, nullptr, Args, CxtI);
-      else if (any_of(enumerate(NMask), [&](const auto &ME) {
```

Comment:
Nit: perhaps add the description/explanation why we are not counting already inserted elements to the comment above?